vscode-extension/src/languageClient.ts

```diff
@@ -70,13 +72,32 @@ export async function createAndStartClient(context: RelayExtensionContext) {
   };
 
   // Create the language client and start the client.
-  context.client = new LanguageClient(
+  const client = new LanguageClient(
     'RelayLanguageClient',
     'Relay Language Client',
     serverOptions,
     clientOptions,
   );
 
+  let showStatusDisposable: Disposable | undefined;
+  client.registerFeature({
+    fillClientCapabilities() {},
+
+    initialize() {
+      showStatusDisposable = client?.onRequest(
+        new RequestType<ShowStatusParams, void, void>('window/showStatus'),
+        (params) => {
+          handleShowStatusMethod(context, params);
+        },
+      );
+    },
+
+    dispose() {
+      showStatusDisposable?.dispose();
+    },
+  });
```

Comment:
See my comment in `statusBar.ts` I wonder if we could move this logic into that file.